src/runtime/server/middleware/cspNonceHandler.ts

```diff
@@ -0,0 +1,46 @@
+import crypto from 'node:crypto'
+import { createError, defineEventHandler, getCookie, sendError, setCookie } from 'h3'
+import { getRouteRules } from '#imports'
+
+export type NonceOptions = {
+  enabled: boolean;
+  mode: 'renew' | 'check';
+  value: undefined | (() => string);
+}
+
+export default defineEventHandler((event) => {
+  let csp = `${event.node.res.getHeader('Content-Security-Policy')}`
+  const routeRules = getRouteRules(event)
+
+  if (routeRules.security.nonce !== false) {
+    const nonceConfig: NonceOptions = routeRules.security.nonce
+
+    // See if we are checking the nonce against the current value, or if we are renewing the nonce value
+    let nonce: string | undefined
+    switch (nonceConfig?.mode) {
+      case 'check': {
+        nonce = getCookie(event, 'nonce')
+
+        if (!nonce) {
+          return sendError(event, createError({ statusCode: 401, statusMessage: 'Nonce is not set' }))
+        }
+
+        break
+      }
+      case 'renew':
+      default: {
+        nonce = nonceConfig?.value ? nonceConfig.value() : Buffer.from(crypto.randomUUID()).toString('base64')
+        setCookie(event, 'nonce', nonce, { sameSite: true, secure: true })
```

Comment:
What about if site will not use https connection? Maybe it would be better to look at strictTransportSecurity option in NuxtSecurity config?

Reply:
Won't it interfere with development mode (with localhost without https)? Some alert should be shown, I think, because it is not transparent for understanding in currect sulution (or looking at strictTransportSecurity option).

Reply:
To be honest, I'm hoping that any users of this library will not be serving a webapp without SSL in this day and age.

I would even then suggest to just add a middleware to this plugin which checks for:

* if production and not on ssl -> throw warning about security without SSL being pointless

That being said, it seems that secure: true cookies are currently transmitted on localhost domains (probably to developers).

Reply:
I agree with you that ssl is must have))

But "if production and not on ssl" - if site will work inside docker container and proxy without ssl - checking will failed too.

I think `secure` might be with true value, as you have done)